scripts/experiments_images/ocsvm.jl

```diff
@@ -0,0 +1,112 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+using GenerativeAD.Models
+using BSON
+using StatsBase: fit!, predict, sample
+
+using MLDataPattern
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+		required = true
+		arg_type = Int
+		help = "seed"
+	"dataset"
+		required = true
+		arg_type = String
+		help = "dataset"
+	"anomaly_classes"
+		arg_type = Int
+		default = 10
+		help = "number of anomaly classes"
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed, anomaly_classes = parsed_args
+
+modelname = "OCSVM"
+
+function sample_params()
+	par_vec = (round.([10^x for x in -4:0.1:2],digits=5),["poly", "rbf", "sigmoid"])
+	argnames = (:gamma,:kernel)
+	return (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
+end
+
+function joint_fit(models, data_splits)
+    info = []
+    for (model, data) in zip(models, data_splits)
+        push!(info, fit!(model, data))
+    end
+    return info
+end
+
+function joint_prediction(models, data)
+    joint_pred = Array{Float32}(undef, length(models), size(data,2))
+    for (i,model) in enumerate(models)
+        joint_pred[i,:] = predict(model, data)
+    end
+    return Flux.mean(joint_pred, dims=1)
+end
+
+function fit(data, parameters, n_models=10)
+    # construct model - constructor should only accept kwargs
+    models = [GenerativeAD.Models.OCSVM(;parameters...) for _ = 1:n_models]
+	
```

Comment:
What I would do is that I would reshape all data here, e.g. `data[1][1] = reshape(data[1][1], :, size(data[1][1],4))` (the same for (val and tst data).

Reply:
Then the rest is gonna work as you have it written already.